lib/setupmanagers/qemuhck/devices/iommu.rb

```diff
@@ -0,0 +1,31 @@
+# typed: true
+# frozen_string_literal: true
+
+module AutoHCK
+  class QemuHCK
+    module Devices
+      extend T::Sig
+
+      sig { params(logger: AutoHCK::MultiLogger).returns(Models::QemuHCKDevice) }
+      def self.iommu(logger)
+        cpuinfo = File.read('/proc/cpuinfo')
+        vendor = T.must(cpuinfo.match(/^vendor_id\s*:\s*(\S+)/))[1]
```

Comment:
Check if `cpuinfo.match(/^vendor_id\s*:\s*(\S+)/)` if `nil` just as you do for `[1]` below instead of calling `T.must`.